arviz/data/inference_data.py

```diff
@@ -91,7 +91,15 @@ def to_netcdf(self, filename, compress=True):
         return filename
 
     def __add__(self, other):
-        """Concatenate two InferenceData objects."""
+        """Concatenate two InferenceData objects.
+
+        example
+        -------
+        A._groups == ["posterior", "posterior_predictive"]
```

Comment:
It's usually not good to add an example that uses a "protected" class. The underscore suggests that users should not be using the attribute directly, but then the docstring encourages them to.

I think it would be better to do have an example that avoids adding groups by direct assignment. My suggestion would be to use `az.convert_to_inference_data` to generate some simulated data/InferenceData objects and combine those

`https://arviz-devs.github.io/arviz/generated/arviz.convert_to_inference_data.html#arviz.convert_to_inference_data

Reply:
I understand that part but when someone is concatenating something how come he knows that which groups does that particular object of inference data belongs without showing him?

Reply:
## Will this example for __add__ be alright?
---> A
Inference data with groups:
	> posterior
---> emcee_data
Inference data with groups:
	> observed_data
---> C=A+emcee_data
---> C
Inference data with groups:
	> posterior
	> observed_data

Reply:
Yes, that looks much better!